src/webview/serverless-function/app/createFunction.tsx

```diff
@@ -0,0 +1,332 @@
+                    templates: ['Cloud Events', 'Flask', 'HTTP', 'WSGI']
+                })
+            }
+            this.setState({
+                language: e.target.value,
+                template: ''
+            });
+        } else {
+            this.setState({
+                template: e.target.value
+            });
+        }
+    }
+
+    handleCreateBtnDisable(): boolean {
+        return !this.state.functionData || this.state.functionData.name.length === 0 || this.state.functionData.error
+            || !this.state.imageData || this.state.imageData.name.length === 0 || this.state.imageData.error
+            || !this.state.wsFolderPath || this.state.language.length === 0 || this.state.template.length === 0;
+    }
+
+    createFunction = (): void => {
+        this.props.onCreateSubmit(this.state.functionData.name,
+            this.convert(this.state.language), this.convert(this.state.template), this.state.wsFolderPath, this.state.imageData.name)
+    }
+
+    render(): React.ReactNode {
+        const { wsFolderItems, wsFolderPath, functionData, templates, language, template, images, imageData } = this.state;
+        const languages = ['Go', 'Node', 'Python', 'Quarkus', 'Rust', 'Spring Boot', 'TypeScript'];
+        const folders: any[] = ['New Folder'];
+        folders.push(...wsFolderItems);
+        const filter = createFilterOptions<string>();
+        const imageRegex = RegExp('[^/]+\\.[^/.]+\\/([^/.]+)(?:\\/[\\w\\s._-]*([\\w\\s._-]))*(?::[a-z0-9\\.-]+)?$');
+        return (
+            <>
+                <LoadModal show={this.props.loadScreen} />
+                <FormControl sx={{ margin: '2rem 0 0 2rem' }}>
+                    <TextField
+                        type='string'
+                        variant='outlined'
+                        required
+                        defaultValue={functionData.name}
+                        error={functionData.error}
+                        onChange={(e) => this.validateName(e.target.value)}
+                        id='function-name'
+                        placeholder='Function Name *'
+                        sx={{
+                            input: {
+                                color: 'var(--vscode-settings-textInputForeground)',
+                                height: '5px !important',
+                                '&::placeholder': {
+                                    color: 'var(--vscode-settings-textInputForeground) !important',
+                                    opacity: '1 !important'
+                                }
+                            }
+                        }}
+                        helperText={functionData.helpText}
+                    />
+                </FormControl>
+                <FormControl sx={{ margin: '2rem 0 0 2rem', width: 'auto' }}>
+                    <Autocomplete
+                        value={imageData.name}
+                        disabled={functionData.error || functionData.name.length === 0}
```

Comment:
Why do we need this check ? A user can first opt to provide the build image and then provide the name of function. Both actions are mutually exclusive. We should remove the check of function name length.